src/Amalgam/interpreter/InterpreterOpcodesEntityControl.cpp

```diff
@@ -197,11 +197,7 @@ EvaluableNodeReference Interpreter::InterpretNode_ENT_ASSIGN_ENTITY_ROOTS_and_AC
 			}
 		}
 
-	#ifdef MULTITHREAD_SUPPORT
-		target_entity->CollectGarbage(&memoryModificationLock);
-	#else
-		target_entity->CollectGarbage();
-	#endif
+		target_entity->CollectGarbageWithEntityWriteReference();
```

Comment:
So it seems like the difference is basically we used to hold a read lock during this garbage collection; with this change that is no longer the case. But why is it ok to not hold that lock here?

Reply:
Oh, I see, target_entiy is a WriteReference which itself has a lock. And since GarbageCollection is specific to one EvaluableNode, that is sufficient. Do I have that right?